File: Hms/Controllers/RateController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using BusinessLayer;
using DataModel;
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;

namespace Hms.Controllers
{
    public class RateController : Controller
    {
        private RateBL rateBL = new RateBL();
        private AgentBL agentBL = new AgentBL();
        private PlanBL planBL = new PlanBL();
        private RoomTypeBL roomTypeBL = new RoomTypeBL();
        public ActionResult Index()
        {
            if (!rateBL.CheckExists())
            {
                rateBL.InsertDefaults();
            }
            ViewBag.Title = "Rate";
            ViewBag.ControllerName = "Rate";
            return View(rateBL.GetRateList());
        }

        [HttpPost]
        public JsonResult UpdateRateValue(string data)
        {
            int id, amount, exppercharge;
            var obj = (JObject)JsonConvert.DeserializeObject(data);
            id = Convert.ToInt32(obj["id"]);
            amount = Convert.ToInt32(obj["amount"]);
            exppercharge = Convert.ToInt32(obj["expcharge"]);
            rateBL.UpdateValue(id, amount, exppercharge);
            return Json(1);
        }

        public JsonResult GetRateList()
        {
            return new JsonResult { Data = rateBL.GetRateList(), JsonRequestBehavior = JsonRequestBehavior.AllowGet };
        }

        public JsonResult GetAgentList()
        {
            return new JsonResult { Data = agentBL.ListAgent(), JsonRequestBehavior = JsonRequestBehavior.AllowGet };
        }

        public JsonResult GetPlanList()
        {
            return new JsonResult { Data = planBL.ListPlan(), JsonRequestBehavior = JsonRequestBehavior.AllowGet };
        }

        public JsonResult GetRoomTypeList()
        {
            return new JsonResult { Data = roomTypeBL.ListRoomType(), JsonRequestBehavior = JsonRequestBehavior.AllowGet };
        }
    }
}

File: Hms/Scripts/RateController/RateController.js
var MyApp = angular.module('MyApp',[])
    .controller('RateController', function ($scope, $http, RateService) {
        RateService.GetAgentLists().then(function (d){
            $scope.AgentList = d.data;
        });
        RateService.GetPlanLists().then(function (d) {
            $scope.PlanList = d.data;
        });
        RateService.GetRoomTypeLists().then(function (d) {
            $scope.RoomTypeList = d.data;
        });
        RateService.GetRateList().then(function (d) {
            $scope.RateList = d.data;
        });
        $scope.UpdateRate = (function (index) {
            var id = $scope.RateList[index].Id;
            var amount = $scope.RateList[index].Amount;
            var expCharge = $scope.RateList[index].ExtraPersonCharge;
            var data = {"id":id,"amount":amount,"expcharge":expCharge};
            
            RateService.UpdateRateValue(JSON.stringify(data)).then(function (d) {
                RateService.GetRateList().then(function (c) {
                    $scope.RateList = c.data;
                });
            });
        });
    })
    .factory('RateService', function ($http) {
        var fac = {};
        fac.GetAgentLists = function () {
            return $http.get('/Rate/GetAgentList');
        };

        fac.GetPlanLists = function () {
            return $http.get('/Rate/GetPlanList');
        }

        fac.GetRoomTypeLists = function () {
            return $http.get('/Rate/GetRoomTypeList');
        }

        fac.GetRateList = function () {
            return $http.get('/Rate/GetRateList');
        }

        fac.UpdateRateValue = function (data) {
            console.log(data);
            return $http.post('/Rate/UpdateRateValue/?data='+ data);
        }
        return fac;
    });